js/GroupedBarChart.js

```diff
@@ -120,13 +114,67 @@ var GroupedBarChart = function(param)
             {
               // Ask about YW
             }
-
             return sort ? _.sortBy(values,
                 function(d)
                 {
                     return that.toDate(d.xval) - 0;
                 }) : values;
+        },
+        initChart : function(data) {
+          var that = this;
+          //get unique xvals from the data, these are the main categories. they make up the x0 domain
+          //use pluck and unique
+          var mainCategories = _.uniq(_.pluck(data, 'xval'));
+          console.log(mainCategories);
+
+          var maxY = _.max(_.pluck(data, 'yval'));
+          console.log(maxY);
+
+          //get the max number of subcategories, or count the number of names in  each
+          //
+          //maybe later  make it dynamic (3 instead of four)
+          var subCategories = _.uniq(_.pluck(data, 'name'));
```

Comment:
Since you're just passing the raw data now to initChart, do we still need setGroupedBarData() ?